src/Components/Blazor/Build/src/targets/Blazor.MonoRuntime.targets

```diff
@@ -611,28 +611,20 @@
     Inputs="$(BlazorBuildBootJsonInputsCache);@(_BlazorDependencyInput)"
     Outputs="$(BlazorBootJsonIntermediateOutputPath)">
     <ItemGroup>
-      <_UnlinkedAppReferencesPaths Include="@(_BlazorDependencyInput)" />
       <_AppReferences Include="@(BlazorItemOutput->WithMetadataValue('Type','Assembly')->WithMetadataValue('PrimaryOutput','')->'%(FileName)%(Extension)')" />
       <_AppReferences Include="@(BlazorItemOutput->WithMetadataValue('Type','Pdb')->'%(FileName)%(Extension)')" Condition="'$(BlazorEnableDebugging)' == 'true'" />
     </ItemGroup>
     <PropertyGroup>
       <_LinkerEnabledFlag Condition="'$(_BlazorShouldLinkApplicationAssemblies)' != ''">--linker-enabled</_LinkerEnabledFlag>
       <_ReferencesArg Condition="'@(_AppReferences)' != ''">--references &quot;$(BlazorBootJsonReferencesFilePath)&quot;</_ReferencesArg>
-      <_EmbeddedResourcesArg Condition="'@(_UnlinkedAppReferencesPaths)' != ''">--embedded-resources &quot;$(BlazorEmbeddedResourcesConfigFilePath)&quot;</_EmbeddedResourcesArg>
     </PropertyGroup>
 
     <WriteLinesToFile
       File="$(BlazorBootJsonReferencesFilePath)"
       Lines="@(_AppReferences)"
       Overwrite="true" />
 
-    <WriteLinesToFile
-      Condition="'@(_UnlinkedAppReferencesPaths)' != ''"
-      File="$(BlazorEmbeddedResourcesConfigFilePath)"
-      Lines="@(_UnlinkedAppReferencesPaths)"
-      Overwrite="true" />
-
-    <Exec Command="$(BlazorBuildExe) write-boot-json &quot;@(IntermediateAssembly)&quot; $(_ReferencesArg) $(_EmbeddedResourcesArg) $(_LinkerEnabledFlag) --output &quot;$(BlazorBootJsonIntermediateOutputPath)&quot;" />
+    <Exec Command="$(BlazorBuildExe) write-boot-json &quot;@(IntermediateAssembly)&quot; $(_ReferencesArg) $(_LinkerEnabledFlag) --output &quot;$(BlazorBootJsonIntermediateOutputPath)&quot;" />
```

Comment:
With the reduced set of dependencies this can now probably be an MSBuild task. This significantly can reduce the amount of ceremony here and will improve error reporting.

Reply:
Yup. That's the plan - https://github.com/aspnet/AspNetCore/pull/17157